label_studio/tasks/models.py

```diff
@@ -117,24 +185,28 @@ def file_upload_name(self):
 
     @classmethod
     def get_locked_by(cls, user, project=None, tasks=None):
-        """ Retrieve the task locked by specified user. Returns None if the specified user didn't lock anything.
-        """
+        """Retrieve the task locked by specified user. Returns None if the specified user didn't lock anything."""
         lock = None
         if project is not None:
-            lock = TaskLock.objects.filter(user=user, expire_at__gt=now(), task__project=project).first()
+            lock = TaskLock.objects.filter(
+                user=user, expire_at__gt=now(), task__project=project
+            ).first()
         elif tasks is not None:
-            locked_tasks = tasks.filter(locks__user=user, locks__expire_at__gt=now())[:1]
-            if locked_tasks:
-                return locked_tasks[0]
```

Comment:
made a little improvement here by using `first()`

Reply:
It's better to be very accurate with first(), we had several situations with huge slowdowns because of first():

https://www.notion.so/humansignal/Backend-Handbook-ce7324a497cb4b40b91911a429e34fcb?pvs=4#2f4ad3164b5a4c8ea07705c8584895ae